server/controllers/industry.js

```diff
@@ -141,8 +216,191 @@ const deleteJob = async (req, res) => {
+    }
+};
+
+const createAssessment = async (req, res) => {
+    try {
+        const accessToken = req.header('Authorization').split(' ')[1];
+        const decodeAccessToken = jwt.verify(accessToken, process.env.SECRET_ACCESS_TOKEN);
+
+        //check if user exists
+        const user = await retrieveOrCacheUsers(req, decodeAccessToken.email); 
+
+        // check if user type is industry
+        if (user.userType.Type != process.env.INDUSTRY) {
+            return res.status(400).json(generateRes(false, 400, 'BAD_REQUEST', {}));
+        }
+
+        const assessmentName = req.body.name;
+        const assessmentQuestions = req.body.questions;
+
+        const industryData = (!user.userType.industryData)
+            ? new IndustryData({
+                jobs: {
+                    active: [],
+                    draft: [],
+                    archived: [],
+                },
+                assessments: [],
+            })
+            : await retrieveOrCacheIndustry(req, user.userType.industryData);
+
+        industryData.assessments.push({
+            name: assessmentName,
+            questions: assessmentQuestions,
+        });
+        await industryData.save();
+
+        if (!user.userType.industryData) {
+            user.userType.industryData = industryData._id;
+            await user.save();
+        }
+
+        return res.status(201).json(generateRes(true, 201, "ASSESSMENT_CREATED", {}));
+    } catch (error) {
+        return res.status(400).json(generateRes(false, 400, "BAD_REQUEST", { error }));
+    }
+};
+
+const editAssessment = async (req, res) => {
+    try {
+        const accessToken = req.header('Authorization').split(' ')[1];
+        const decodeAccessToken = jwt.verify(accessToken, process.env.SECRET_ACCESS_TOKEN);
+
+        //check if user exists
+        const user = await retrieveOrCacheUsers(req, decodeAccessToken.email); 
+
+        // check if user type is industry
+        if (user.userType.Type != process.env.INDUSTRY) {
+            return res.status(400).json(generateRes(false, 400, "BAD_REQUEST", {}));
+        }
+
+        const assessmentId = req.body.assessmentId;
+        const assessmentDetails = req.body.assessmentDetails;
+
+        // find the assessment with the provided ID and set its name and questions
+        // to the provided values
+        const result = await IndustryData.updateOne(
+            { _id: user.userType.industryData, 'assessments._id': assessmentId},
+            { $set: {
+                'assessments.$.name': assessmentDetails.name,
+                'assessments.$.questions': assessmentDetails.questions,
+            }},
+        );
+        if (result.acknowledged) {
+            return res.status(200).json(generateRes(true, 200, "ASSESSMENT_UPDATED", {}));
+        }
+
+        return res.status(400).json(generateRes(false, 400, "BAD_REQUEST", {}));
+    } catch (error) {
+        return res.status(400).json(generateRes(false, 400, "BAD_REQUEST", { error }));
```

Comment:
These catch error responses should all be server_error 500 I believe, since the controller caught an error. I have updated most (if not all) for the faculty and student controllers so i'd recommend doing the same for each here